### ISSUE TITLE
Marking a private `Enum` member as `member()` does not work

### ISSUE THREAD

**Issue by sobolevn** [python/mypy#18563]
Consider this example:

```python
from enum import Enum, member

class Pet(Enum):
    __CAT = member(1)

reveal_type(Pet.__CAT)  # N: Revealed type is "enum.member[Literal[1]?]"
```

Link: https://mypy-play.net/?mypy=latest&python=3.12&gist=63b056215c22ba7ff09e2559dd2ae08a

However, in runtimes it fails with `AttributeError`, because `__CAT` is mangled to `_Pet__CAT` as regular private names are.

We need to fix this corner-case.

Refs https://github.com/python/mypy/issues/18491
Refs https://github.com/python/mypy/issues/18557

**Comment by sobolevn** [comment]
I also think that the revealed type should be just `Literal[1]` ...?

**Comment by tyralla** [comment]
PR #16715 deals with name mangling.  Would it fix this issue as well?

**Comment by sobolevn** [comment]
Yes, this problem is not related to `Enum`. It happens for all similar case:

```python
from enum import Enum

class A:
    def __init__(self) -> None:
        self.__private = 1
        
A().__private  # false positive
```
